src/Microsoft.Identity.Client/Azure/Constants.cs

```diff
@@ -0,0 +1,173 @@
+        ///     store
+        /// </summary>
+        public const string AzureCertificateStoreEnvName = "AZURE_CERTIFICATE_STORE";
+
+        /// <summary>
+        ///     AzureCertificateStoreLocationEnvName is the name of the environment variable which holds the location of the
+        ///     certificate store (CurrentUser or LocalMachine)
+        /// </summary>
+        public const string AzureCertificateStoreLocation = "AZURE_CERTIFICATE_STORE_LOCATION";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on AppService
+        /// </summary>
+        public const string ManagedIdentityAppServiceApiVersion = "2017-08-01";
+
+        /// <summary>
+        /// API Version used for the Managed Identity metadata endpoint on Virtual Machines
+        /// </summary>
+        public const string ManagedIdentityVMApiVersion = "2018-02-01";
+
+        /// <summary>
+        /// Managed Identity loopback IP address
+        /// </summary>
+        public const string ManagedIdentityLoopbackAddress = "169.254.169.254";
+
+        /// <summary>
+        /// Managed Identity metadata endpoint
+        /// </summary>
+        public const string ManagedIdentityMetadataEndpoint = "http://" + ManagedIdentityLoopbackAddress + "/metadata";
+
+        /// <summary>
+        /// Managed Identity 
+        /// </summary>
+        public const string ManagedIdentityTokenEndpoint = ManagedIdentityMetadataEndpoint + "/identity/oauth2/token";
+
+        /// <summary>
+        /// ManagedIdentityEndpointEnvName is the name of the environment variable which holds the endpoint for the
+        /// managed identity service running in Azure AppService 
+        /// </summary>
+        public const string ManagedIdentityEndpointEnvName = "MSI_ENDPOINT";
+
+        /// <summary>
+        /// ManagedIdentitySecretEnvName is the name of the environment variable which holds the secret for use in
+        /// calling the managed identity service running in Azure AppService
+        /// </summary>
+        public const string ManagedIdentitySecretEnvName = "MSI_SECRET";
+
+        /// <summary>
+        /// AadAuthorityEnvName is the name of the environment variable which holds the hostname of the service token service (STS) from which MSAL.NET will acquire the tokens. Ex. login.microsoftonline.com
+        /// </summary>
+        public const string AadAuthorityEnvName = "AAD_AUTHORITY";
+    }
+
+    /// <summary>
+    /// ErrorCodes contains string error codes that can be emitted from this namespace
+    /// </summary>
+    public static class ErrorCodes
+    {
+        /// <summary>
+        /// FailedParseOfManagedIdentityExpiration happens when we can't parse the expiry on the token
+        /// </summary>
+        public const string FailedParseOfManagedIdentityExpiration = "failed_parse_of_managed_identity_token_expiry";
```

Comment:
wouldn't it be better that these be members of the exception classes that throw them?